src/invocation/InvocationService.ts

```diff
@@ -162,3 +170,44 @@ export class InvocationService {
     }
 }
 
+export class ListenerService {
+    private client: HazelcastClient;
+    private listenerIdToCorrelation: { [id: string]: Long} = {};
+    private internalEventEmitter: EventEmitter;
+
+    constructor(client: HazelcastClient) {
+        this.client = client;
+        this.internalEventEmitter = new EventEmitter();
+        this.internalEventEmitter.setMaxListeners(0);
+    }
+
+    registerListener(codec: any, handler: any): Q.Promise<string> {
+        var deferred = Q.defer<string>();
+        var invocation = new Invocation(codec.encodeRequest(true));
+        invocation.handler = handler;
+        var listenerIdToCorrelation = this.listenerIdToCorrelation;
+        this.client.getInvocationService().invoke(invocation).then(function(responseMessage) {
```

Comment:
If you use a "fat arrow" syntax (=>) then you don't need to bother with "this" reference reassigning